### ISSUE TITLE
Confusing and Unnecessary use of IIFE's causing errors on correct answers - ES6 

### ISSUE THREAD

**Issue by AnonymousLords** [freeCodeCamp/freeCodeCamp#17507]
* Browser Name: Chrome
* Browser Version: Chrome 67
* Operating System: Windows 10

I have started the new ES6 curriculum of freeCodeCamp in hopes of simplifying my ES5 ways. I have recently come across unnecessary, confusing, unlikely use of the Immediately-invoked function expression's that will cause users not to past the test even though their code is correct.

The IIFE first appears in the challenge "Set Default Parameters for Your Functions" with seemingly no explanation, or reasoning on why it is their.

--> https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/set-default-parameters-for-your-functions/

This will confuse new coders, and i have seen people just copy and paste the working code inside of the IIFE and passing have no idea what it is actually doing (bad practice). Even if they did know what it is doing, its a very limited and not as widely used technique of formatting that it shouldn't even be in this challenge.

The twist on this is the fact that the IIFE isn't even required to pass the challenge:

```
"use strict";
const increment = (number, value = 1) => number+value;
increment(10, 1)
```
--> https://repl.it/@John_Nicole/ES6-9-Set-Default-Parameters-for-Your-Functions

This proves how confusing and varying the challenge can possibly be when you can start to infer its not important with no idea what it is. 

But wait, theirs more:

If you go to the next challenge "Use the Rest Operator with Function Parameters", the IIFE appears again.
--> https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/use-the-rest-operator-with-function-parameters

Only this time, the IIFE is required to pass the test even though its never clearly specified what the IIFE actually is, doing, or even that it is needing. 

The passing code for the challenge is:

```
const sum = (function () {
    const sum = (...args) => {
      return args.length > 0 ? args.reduce((num, i) => num+=i) : 0
     }
  return sum;
})();
```
You can easily accomplish the same task with:
```
const sum = (...args) => {
return args.length > 0 ? args.reduce((num, i) => num+=i) : 0
// num is accumulator, i is the number.
}
```
--> https://repl.it/@John_Nicole/ES6-10-Use-the-rest-operator
(Use ctrl+/ to uncomment and comment).

Yet when i run this code i get that this requirement is not met:

``` X, The sum function uses the ... spread operator on the args parameter.```

When it is clearly used. Even if the IIFE is not at fault for this it still showed be removed. From what i have been told, the IIFE not being present is the cause with seemingly no explanation by the challenge.

You can see how the challenges can vary way too much to the point where getting the same output in a way that meets the requirements can even be marked as wrong. Its is unfair, and confusing.

There is even another example of confusing answers to ES6 challanges

Go back to the challenge "Use Arrow Functions to Write Concise Anonymous Functions"
--> https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/use-arrow-functions-to-write-concise-anonymous-functions

Trying to teach arrow functions they give you this:
```
var magic = function() {
  "use strict";
  return new Date();
};
```
…And ask you to first do arrow code, also to change var to const

Yes the purpose is to teach you arrow code functions, but its a bad example as some people could look at the challenge differently, since the point of ES6 is simple code.

```const magic = new Date()```

Yes this code doesn't past the challenge, but it is correct in the sense that they both would do the same thing, one is simpler.

Correct answer (just using a unnecessary function):

```const magic = () =>  new Date()```

To recap we have seen:

1. Unnecessary use of IIFE's
2. Challenges not  accepting seemingly and logically correct answers 
3. Confusing ways of teaching something seemingly simple.

If you are going to put IIFE's as mandatory in challenges, at least explain it in its own challenge.
